scripts/dprpwg-gen.py

```diff
@@ -0,0 +1,77 @@
+#! /usr/bin/env python3
+#
+# This script generates a dprpwg_config.h file from dprpwg_config.stubs.h by
+# using pseudo-random numbers. The numbers are generated accourding to the
+# python PEP506 Note. As such, they SHOULD be suitable for cryptographic use.
+
+
+import argparse
+import ctypes
+import secrets
+import sys
+from pathlib import Path
+
+THIS_DIR = Path(__file__).resolve()
+DEFAULT_CONFIG = THIS_DIR.parent.parent / "src" / "dprpwg_config.stub.h"
+
+class Help:
+    TEMPLATE = """Path to the dprpwg_config.sub.h file to be used as a template
+to generate a suitable dprpwg_config.h file"""
+    OUTPUT = """Path to the output header to generate. If not provided, stdout
+will be used"""
+
+def getopts(argv):
+    parser = argparse.ArgumentParser(description='DPRPWG config.h generator')
+    parser.add_argument("--template", "-t", metavar='CONFIG.h',
+                        default=DEFAULT_CONFIG, help=Help.TEMPLATE)
+    parser.add_argument("--output", "-o", metavar='OUTPUT.h',
+                        help=Help.OUTPUT)
+    return parser.parse_args(argv[1:])
+
+def main(argv):
+    args = getopts(argv)
+
+    # Here is the list of all the #define values we shall substitute
+    variables = ("PW_MUL", "PW_SEEK_MUL", "PW_INV_MUL",
+                 "DOM_MUL", "DOM_SEEK_MUL", "DOM_INV_MUL",
+                 "YR_MUL", "YR_SEEK_MUL", "YR_INV_MUL")
+
+    # The values to substitute are 'unsigned int'. So, determine the count of
+    # bits that can be used in the generation of the random values.
+    # We assume 8 bits per byte.
+    bits = ctypes.sizeof(ctypes.c_uint * 8)
+
+    # Make a substitution table that to each variable associates a random
+    # number (as a string)
+    replace_table = {x: str(secrets.randbits(bits)) for x in variables}
+
+    def _find_variable(text_line):
+        """Lambda: if one the variables described in the scope above is
+        contained in 'text_line', we will return the value it shall be replaced
+        by.
+        """
+        for variable in variables:
+            if variable in text_line:
+                return replace_table[variable]
+
+    text = ""
+
+    # Go through each line of the file. If a line contains the variable, we
+    # replace "<YOUR_NUMBER>" by the random number associated with the variable
+    with open(args.template, "r") as filep:
+        for line in filep:
+            var = _find_variable(line)
+            if var is not None:
+                line = line.replace("<YOUR_NUMBER>", var)
+            text += line
+
+    # Finally, write the config.h header to the filesystem. Shall be be in the
+    # specified path or in the stdout otherwise.
+    if args.output is not None:
+        with open(args.output, "w") as filep:
+            filep.write(text)
```

Comment:
Yes, generating the `dprpwg_config.h` is painful, but you should do it once, and **you absolutely do not want to overwrite your own file once it has been generated**. Now I think it is far too easy...

Maybe:
- add a check to avoid replacing an existing file;
- make the output file read-only after generation;
- remove the `stdout` output

Reply:
Okay

Reply:
Please note that it cannot  prevent the file from being removed.